### PR TITLE
Troubleshooting - Add State Lock error

### PR DESCRIPTION
**Pull request by felipecrescencio-cit**
Added Error acquiring the state lock section and details on reviewing terraform state.
Fixes part of the issue [488](https://github.com/terraform-google-modules/terraform-example-foundation/issues/488#issue-915572280).

### CODE REVIEW

File: docs/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -256,8 +261,51 @@ Most likely this is related to a billing quota issue.
...
+
+**Error message:**
+
+```text
+Error: Error acquiring the state lock
+```
+
+**Cause:**
+
+This message means you are trying to execute a Terraform command with the [Terraform State locked](https://www.terraform.io/language/state/locking).
```
Comment (bharathkkb) [inline docs/TROUBLESHOOTING.md:280]:
```suggestion
This message means that you are trying to apply a Terraform configuration with a remote backend that is in a [locked state](https://www.terraform.io/language/state/locking).
```

File: docs/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -256,8 +261,51 @@ Most likely this is related to a billing quota issue.
...
+
+If the Terraform process was unable to finish due to an unexpected event, i.e build timeout or terraform process killed. It will keep the Terraform State **locked**.
+
+**Solution:**
+
+Keep in mind that after removing the Terraform State lock you will need to [review and manipulate](https://www.terraform.io/cli/state#manipulating-terraform-state) your Terraform State and maybe [import](https://www.terraform.io/cli/import#import) some resources.
```
Comment (bharathkkb) [inline docs/TROUBLESHOOTING.md:286]:
```suggestion
Keep in mind that after removing the Terraform State lock you may need to [review and manipulate](https://www.terraform.io/cli/state#manipulating-terraform-state) your Terraform State and maybe [import](https://www.terraform.io/cli/import#import) some resources.
```

File: docs/TROUBLESHOOTING.md [hunk 3]
```diff
@@ -256,8 +261,51 @@ Most likely this is related to a billing quota issue.
...
+
+Here are some sample commands on how to remove the Terraform State lock.
```
Comment (bharathkkb) [inline docs/TROUBLESHOOTING.md:288]: I would rephrase this in the context of unlocking a single root config like https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/2-environments/envs/development. So you can guide the user through git clone repo, cd to env, change backend if needed, run tf locally to confirm state is locked, locally unlock state, do a plan to validate no diff or look at import/state rm for massage the state.

File: docs/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -256,8 +261,51 @@ Most likely this is related to a billing quota issue.
...
+
+```bash
+terraform force-unlock $(terraform apply 2>&1 > /dev/null | grep "ID:" | sed -e 's/.*[^0-9]\([0-9]\+\)[^0-9]*$/\1/')
+```
+
+**Notes:**
+
+- The remove Terraform State lock command **does not** change anything in your state.
+- If you got a Terraform State locked error **most of the time** the Terraform State will be inconsistent.
+- It is a **strong recommendation** to review the Terraform State before trying to continue on the step you stopped.
```
Comment (bharathkkb) [inline docs/TROUBLESHOOTING.md:311]: I would link to upstream docs rather than having this info here
https://www.terraform.io/language/state/locking#force-unlock

File: docs/TROUBLESHOOTING.md (hunk 3 above)
Comment (bharathkkb) [reply docs/TROUBLESHOOTING.md:288]: Our last instruction could also be to review what caused the lock to happen. If this was due to a build time our recommend increasing build timeout https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/build/cloudbuild-tf-apply.yaml#L15

File: docs/TROUBLESHOOTING.md [hunk 5]
```diff
@@ -256,8 +261,51 @@ Most likely this is related to a billing quota issue.
...
+cd <YOUR-REPO-DIR>
+export backend_bucket=$(terraform -chdir="<YOUR-FOUNDATION-EXAMPLE-DIR>/0-bootstrap/" output -raw gcs_bucket_tfstate)
+for i in `find -name 'backend.tf'`; do sed -i "s/UPDATE_ME/${backend_bucket}/" $i; done
+terraform init
+```
+
+- Run the command below to remove the Terraform State lock, it will ask for a confirmation and you will be able to run terraform commands again.
+
+```bash
+terraform force-unlock $(terraform apply 2>&1 > /dev/null | grep "ID:" | sed -e 's/.*[^0-9]\([0-9]\+\)[^0-9]*$/\1/')
```
Comment (bharathkkb) [inline docs/TROUBLESHOOTING.md:304]: We should just ask the user to manually get the lock id from the apply they ran to validate that the state is still locked. State unlocking can be dangerous if not carefully reviewed and I am weary of adding more automation.

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for adding this!

File: docs/TROUBLESHOOTING.md (hunk 5 above)
Comment (bharathkkb) [reply docs/TROUBLESHOOTING.md:304]: @felipecrescencio-cit I think we can reword this part

Comment (felipecrescencio-cit) [reply docs/TROUBLESHOOTING.md]: Done

File: docs/TROUBLESHOOTING.md (hunk 4 above)
Comment (felipecrescencio-cit) [reply docs/TROUBLESHOOTING.md]: Done

File: docs/TROUBLESHOOTING.md (hunk 3 above)
Comment (felipecrescencio-cit) [reply docs/TROUBLESHOOTING.md]: done

File: 0-bootstrap/README.md [hunk 6]
```diff
@@ -148,11 +148,12 @@ your current Jenkins manager (controller) environment.
 1. Run `terraform output projects_step_terraform_service_account_email` to get the email address of the admin of step `4-projects`. You need this address in a later procedure.
 1. Run `terraform output cloudbuild_project_id` to get the ID of your Cloud Build project.
 1. Run `terraform output gcs_bucket_tfstate` to get your Google Cloud bucket name from Terraform's state.
-1. Copy the backend:
-   ```
-   cp backend.tf.example backend.tf
+1. Create a `backend.tf` file based on `backend.tf.example` template and update with the name of your Cloud Storage bucket:
```
Comment (bharathkkb) [inline 0-bootstrap/README.md:151]: Lets remove https://github.com/terraform-google-modules/terraform-example-foundation/pull/819/commits/9cb659073a4fd4958825836a0f7f4bdb16542576 from this PR as it is unrelated and addressed in https://github.com/terraform-google-modules/terraform-example-foundation/pull/825

Comment (felipecrescencio-cit) [reply 0-bootstrap/README.md]: Done

**Review by bharathkkb — APPROVED** [review]
